src/winshellext/F3DThumbnailProvider.cxx

```diff
@@ -160,6 +168,15 @@ IFACEMETHODIMP F3DThumbnailProvider::GetThumbnail(UINT cx, HBITMAP* phbmp, WTS_A
   char** argv;
   ::ConvertWCommandLineToArgs(warg, argc, argv);
 
+  // Shell extension CANNOT segfault, we must catch the signal
+  // or it can completelly break explorer.exe
+  signal(SIGTERM, ::SignalHandler);
```

Comment:
Shouldn't we put this in DLL initializing func or restore the default handlers at the end of this function call?

Reply:
> Shouldn't we put this in DLL initializing func 

we could, but the code before cant segfault unless we made a big mistake.

> restore the default handlers at the end of this function call?

Not sure we can recover the default handlers

Reply:
See the doc example. The function signal returns the previous handler. Just restore it at the end.